cpp/include/DataStorm/Node.h

```diff
@@ -38,47 +38,115 @@ namespace DataStorm
         /**
          * Construct a DataStorm node.
          *
-         * A node is the main DataStorm object. It is required to construct topics. The constructor initializes
-         * the Ice communicator using the given arguments. If the communicator creation fails, an Ice exception is
-         * raised.
+         * A node is the main DataStorm object. It is required to construct topics. The node uses the given Ice
```

Comment:
We needed to be able to intercept the construction of the Node's communicator to create a Property set which the "DataStorm" service prefix. So we can no longer just forward everything. @bernardnormier  suggested we just offer a few constructors. This also means we can now always have the custom executor as a trailing parameter.

Reply:
The alternative is to not consider DataStorm a service, which is not really one. It is a user library given that you have to programmatically construct the node.

In theory any servers communicator can be shared with the node and pass to its constructor.

Reply:
I'd prefer making DataStorm an opt-in reserved prefix, as opposed to allowing it all the time like the plugins.

Reply:
Isn't this the same as IceDiscovery and IceLocatorDiscovery which are marked with service-prefix=false.?

Reply:
All the plugins (IceDiscovery etc.) are treated like Ice - always enabled.